kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java

```diff
@@ -91,10 +91,15 @@
 import org.junit.After;
 import org.junit.Test;
 
+import fi.w1.Wpa_supplicant1;
+import fi.w1.wpa_supplicant1.Interface;
+
 public class NMDbusConnectorTest {
 
     private static final String MM_MODEM_BUS_NAME = "org.freedesktop.ModemManager1.Modem";
     private final DBusConnection dbusConnection = mock(DBusConnection.class, RETURNS_SMART_NULLS);
+    private final Wpa_supplicant1 mockedWpaSupplicant = mock(Wpa_supplicant1.class);
+    private final Interface mockedInterface = mock(Interface.class);
```

Comment:
As I said in my previous comment this is not correct. There's only one interface for the entire test.

Please, refer to how we handled the Device here:

https://github.com/eclipse/kura/blob/805c45b5e2ee8b258d192bb589762987494fa7a3/kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java#L121

You should have:
```suggestion
    private final Map<String, Interface> mockedInterfaces = new HashMap<>();
```

and in the `givenMockedDevice` code **create** the mock interface adding it to the above mentioned map for tracking.

Later in the `then` method you'll check the expectation using the "interfaceId" as key to retrieve the mocked object from the map.